addons/sourcemod/scripting/include/json/array.inc

```diff
@@ -809,7 +810,7 @@ methodmap JSON_Array < JSON_Object
                 }
             }
 
-            this.SetHidden(json_size + i, from.GetHidden(i));
+            this.SetHidden(current_size + i, from.GetHidden(i));
```

Comment:
This has nothing to do with the PR, I just noticed it and I think it's wrong. Since it pushed onto the array, the key you are supposed to set hidden must be the length of the array you push onto + index, not the length of the source array?

Reply:
Disregard this. I had inverted the to/from objects in the concat logic. Tests verify this also.

Reply:
Fixed in https://github.com/clugg/sm-json/pull/33.